商品出品機能

nanaiwamoto opened the pull request:
#what
商品出品機能の実装を行いました。
確認をお願いします。

#why
商品出品機能の実装が完了していれば、次の実装に進みたいと思います。
不足している情報があればご教示いただければ幸いです。


[1.ログイン状態の場合は、商品出品ページへ遷移できる動画]
(https://gyazo.com/e16b9a935f587c476541d2d6a39bc7f4)

[2.価格が入力されると同時に、販売手数料と販売利益が表示される動画](https://gyazo.com/142bf86867397f74f94cdbac886790a5)

[3.必要な情報を適切に入力して「出品する」ボタンを押すと、商品情報がデータベースに保存される動画](https://gyazo.com/0fdb4ba35248b0cca14f6480aeb5551a)

[4. 入力に問題がある状態で「出品する」ボタンが押された場合、情報は保存されず、出品ページに戻りエラーメッセージが表示される動画](https://gyazo.com/3a326f992a5a3d43190ca763d39c432d)

[5.ログアウト状態の場合は、商品出品ページへ遷移しようとすると、ログインページへ遷移する動画](https://gyazo.com/a7a932f8f81ef12366c4f382229b7334)

[6.テスト結果の画像](https://gyazo.com/0c3718f9d2cdca325427747070a4397e)

nanaiwamoto committed:
エラーコードがでるように修正

nanaiwamoto committed:
テストコード

yuri240 commented on app/controllers/condition.rb:
```diff
@@ -0,0 +1,14 @@
+class Condition < ActiveHash::Base
```
Conditionはモデルかと思いますので、
コントローラー直下ではなく、適切な場所に配置しましょう！

yuri240 commented on spec/models/item_spec.rb:
```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
+
+RSpec.describe Item, type: :model do
```
異常系テストの項目が不足しているようです。

以下を確認し、不足している項目を追加で実行しましょう！
```
■商品画像
・商品画像が空では出品出来ない
■商品名
・商品名が空では出品できない
■商品説明
・商品説明が空では出品できない
■ActiveHash５項目
・カテゴリーに「---」が選択されている場合は出品できない
・商品の状態に「---」が選択されている場合は出品できない
・配送料の負担に「---」が選択されている場合は出品できない
・発送元の地域に「---」が選択されている場合は出品できない
・発送までの日数に「---」が選択されている場合は出品できない
■価格 
・価格が空では出品できない
・価格に半角数字以外が含まれている場合は出品できない
・価格が300円未満では出品できない
・価格が9_999_999円を超えると出品できない
■出品者情報
・userが紐付いていなければ出品できない
```

yuri240 reviewed (commented):
レビュー依頼ありがとうございます！
コメントさせていただきましたので、ご確認お願いいたします。

**◎次回提出時のお願い2点◎**
```ruby
①プルリクエストのタイトルを「商品出品機能」に変更しましょう。
現状「エラーコードがでるように修正」となっており、何のための機能なのかが判断しにくいためです。

②テストコード修正後のキャプチャ画像を再撮影しましょう。
テストコード変更後は、改めてコマンドの実行とキャプチャ画像の撮影をお願いいたします。
```

ご不明点や疑問点がありましたら、チャット・コネクトメンターにご相談ください。

nanaiwamoto committed:
異常系テスト追加

nanaiwamoto committed:
商品出品機能

ディレクトリの位置変更
app/controllers/condition.rbをcontrollersからモデルへ

nanaiwamoto commented:
確認ありがとうございます。
修正しましたので確認をお願いします。

以下、テストコードの実装画像です。
https://gyazo.com/c872769dd76e90ccfb6bca81832cc6ef

よろしくお願いいたします。

wad-sho commented on app/controllers/items_controller.rb:
```diff
@@ -1,13 +1,36 @@
+  before_action :authenticate_user!, except: [:index, :show]
 
   def index
     @items = Item.order('created_at DESC')
```
@items = Item.order('created_at DESC')

上記コード削除、もしくはコメントアウトをお願いいたします。
こちらは`商品一覧表示機能`の実装内容のためです。

wad-sho commented on app/javascript/item_price.js:
```diff
@@ -0,0 +1,27 @@
+    profitDisplay.textContent = "";
+  } else {
+    const fee = Math.floor(itemPrice * 0.1); // 10%の手数料を計算
+    const profit = itemPrice - fee;         // 利益を計算
```
const profit = itemPrice - fee; 

販売利益算出時にも、Math .floorを用いておきましょう。
価格入力値が小数の際も正しく算出できるようにするためです。

wad-sho reviewed (commented):
実装お疲れ様です！レビューのご依頼ありがとうございます。

コメントいたしましたので、ご確認・修正の後、再度レビューのご依頼お願いいたします！

nanaiwamoto committed:
商品出品機能

1.
@Items = Item.order('created_at DESC')

上記コード削除
2.
const profit = itemPrice - fee;

販売利益算出時にも、Math .floorを用いる

nanaiwamoto commented:
確認ありがとうございます。
修正しましたので確認をお願いします。

yuri240 commented:
早速の修正とレビュー依頼ありがとうございます。

精度高く実装できていることが確認できましたので、
**LGTM**とさせていただきます！

こちらのブランチは忘れずにMergeを行っていただき、
ローカルのMasterにPullしておきましょう。

商品出品機能の実装、お疲れ様でした。
この調子で次の実装も頑張りましょう！応援しております。

nanaiwamoto commented:
ありがとうございます😭
また次回もお願いすることになると思いますが、よろしくお願いいたします‼️
